pkg/sql/catalog/schemadesc/temporary_schema_desc.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schemadesc
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+)
+
+// NewTemporarySchema returns a temporary schema with a given name, id, and
+// parent. Temporary schemas do not have a durable descriptor in the store;
+// they only have a namespace entry to indicate their existence. Given that,
+// a different kind of "synthetic" descriptor is used to indicate temporary
+// schemas.
+//
+// The returned descriptor carries only a basic functionality, requiring the
+// caller to check the SchemaKind to determine how to use the descriptor.
+func NewTemporarySchema(name string, id descpb.ID, parentDB descpb.ID) catalog.SchemaDescriptor {
+	return &temporary{
+		synthetic: synthetic{temporaryBase{}},
+		id:        id,
+		name:      name,
+		parentID:  parentDB,
+	}
+}
+
+// temporary represents the synthetic temporary schema.
+type temporary struct {
+	synthetic
+	id       descpb.ID
+	name     string
+	parentID descpb.ID
+}
+
+var _ catalog.SchemaDescriptor = temporary{}
```

Comment:
Shouldn't this implement `GetPrivileges` properly, at least? Perhaps it's time for a `catalog.Privileges` interface.

Reply:
The same comment applies to `virtual` schemas.